sympy/stats/drv.py

```diff
@@ -123,7 +122,8 @@ def characteristic_function(self, t, **kwargs):
 
     @cacheit
     def compute_moment_generating_function(self, **kwargs):
-        x, t = symbols('x, t', real=True, finite=True, cls=Dummy)
+        t = Dummy('t', real=True)
+        x = Dummy('x', integer=True)
```

Comment:
this does not match the old def...is that intentional?

Reply:
Correct. Discrete distributions. `x` is the index for the summation later, so I am quite sure that it is correct. Also compare with e.g. `compute_quantile` below. Probably one should change in the characteristics functions above as well (didn't notice those until now).